source/common/http/http2/codec_impl.cc

```diff
@@ -399,7 +386,14 @@ int ConnectionImpl::onFrameReceived(const nghttp2_frame* frame) {
     }
 
     case NGHTTP2_HCAT_REQUEST: {
-      stream->decoder_->decodeHeaders(std::move(stream->headers_), stream->remote_end_stream_);
+      if (stream->headers_->Status() && stream->headers_->Status()->value() == "100") {
```

Comment:
nit: This is a little strange given this is a fall through from NGHTTP2_HCAT_RESPONSE. I might not use fall through anymore since the logic is now more complicated, or at least assert this branch is only hit in the NGHTTP2_HCAT_RESPONSE case.